tools/generators/normalize-package-dependencies/index.ts

```diff
@@ -0,0 +1,198 @@
+      return projectConfig.projectType === options.projectType;
+    };
+  }
+  if (options.tag) {
+    active.tag = (projectConfig: ProjectConfiguration) => {
+      return Array.isArray(projectConfig.tags) && projectConfig.tags.includes(options.tag!);
+    };
+  }
+
+  const activeCount = Object.keys(active).length;
+
+  return {
+    hasActive: activeCount > 0,
+    hasActiveMultiple: activeCount > 1,
+    filters: active,
+  };
+}
+
+function shouldBeProjectProcessed(projectConfig: ProjectConfiguration, filters: ReturnType<typeof getActiveFilters>) {
+  const filterPredicates = Object.values(filters.filters);
+
+  if (filters.hasActiveMultiple) {
+    return filterPredicates.every(predicate => predicate(projectConfig));
+  }
+
+  if (filters.hasActive) {
+    return filterPredicates.some(predicate => predicate(projectConfig));
+  }
+
+  return true;
+}
+
+function normalizePackageJsonDependencies(tree: Tree, projectConfig: ProjectConfiguration, graph: ProjectGraph) {
+  const projectDependencies = getProjectDependenciesFromGraph(projectConfig.name!, graph);
+  const packageJsonPath = joinPathFragments(projectConfig.root, 'package.json');
+
+  updateJson(tree, packageJsonPath, json => {
+    updateDepType(json, 'dependencies');
+    updateDepType(json, 'devDependencies');
+
+    return json;
+  });
+
+  return tree;
+
+  function updateDepType(json: PackageJson, depType: 'dependencies' | 'devDependencies' | 'peerDependencies') {
+    const deps = json[depType];
+    if (!deps) {
+      return;
+    }
+
+    for (const packageName in deps) {
+      if (isProjectDependencyAnWorkspaceProject(graph, packageName, projectDependencies)) {
+        deps[packageName] = '*';
+      }
+    }
+  }
+}
+
+function reportPackageJsonDependenciesIssues(issues: ProjectIssues) {
+  const issueEntries = Object.entries(issues);
+
+  if (issueEntries.length === 0) {
+    return;
+  }
+
+  issueEntries.forEach(([projectName, dependencyIssues]) => {
+    logger.log(chalk.bold(chalk.red(`${projectName} has following dependency version issues:`)));
+    // eslint-disable-next-line guard-for-in
+    for (const dep in dependencyIssues) {
+      logger.log(chalk.red(`  - ${dep}`));
+    }
+  });
+
+  logger.info(`All these dependencies version should be specified as '*'`);
+  logger.info(`Fix this by running 'nx workspace-generator normalize-package-dependencies'`);
+
+  throw new Error('package dependency violations found');
+}
+
+function getPackageJsonDependenciesIssues(
+  tree: Tree,
+  projectConfig: ProjectConfiguration,
+  graph: ProjectGraph,
+): Record<string, string> | null {
+  const projectDependencies = getProjectDependenciesFromGraph(projectConfig.name!, graph);
+  const packageJsonPath = joinPathFragments(projectConfig.root, 'package.json');
+  const packageJson = readJson<PackageJson>(tree, packageJsonPath);
+
+  let issues: Record<string, string> | null = null;
+  checkDepType(packageJson, 'dependencies');
```

Comment:
should the generator be able to select what kinds of dependencies to update/check?

Reply:
yeah that will be implemented  as follow-up ( single version policy epic )